java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -61,6 +61,10 @@ public SecretManagerConfigService(@Assisted("projectId") @NonNull String project
 
     @Override
     public void putConfigProperty(ConfigProperty property, String value) {
+        if (property.isEnvVarOnly()) {
```

Comment:
Tip: I like preconditions for these kind of checks.
`Preconditions.checkArgument(!property.isEnvVarOnly(), "Can't put env-only config property: " + property);`